synapse/lib/nexus.py

```diff
@@ -341,13 +341,13 @@ async def eat(self, nexsiden, event, args, kwargs, meta):
         if meta is None:
             meta = {}
 
-        if (nexus := self._nexskids.get(nexsiden)) is None:
-            raise s_exc.NoSuchIden(mesg=f'No Nexus Pusher with iden {nexsiden}.', iden=nexsiden)
+        async with self.cell.nexslock:
+            if (nexus := self._nexskids.get(nexsiden)) is None:
+                raise s_exc.NoSuchIden(mesg=f'No Nexus Pusher with iden {nexsiden}.', iden=nexsiden)
 
-        if event not in nexus._nexshands:
-            raise s_exc.NoSuchName(mesg=f'No Nexus handler for event {event}.', name=event)
+            if event not in nexus._nexshands:
+                raise s_exc.NoSuchName(mesg=f'No Nexus handler for event {event}.', name=event)
 
-        async with self.cell.nexslock:
             self.reqNotReadOnly()
```

Comment:
Should the readonly check be moved up to where the lock is acquired? Does it make more sense to check for readonly before checking if there's a valid handler or is this ordering still ok?

Reply:
Either way is going to blow up, mostly depends on which error we want to take precedence. Feels like the missing pusher/handler should be more rare and should come first but I'm open to flipping it.

Reply:
No need to update. I just noticed the readonly check and thought it might be worth mentioning. I think you're right about wanting to catch the missing nexus handlers first.

Reply:
I think the existing order is fine. A missing handler is more not-okay than something ending up in RO mode. RO mode is also loud once something ends up there.